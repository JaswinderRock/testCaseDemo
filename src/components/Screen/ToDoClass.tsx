import React, { Component } from "react";
import { connect } from "react-redux";
import { setName } from '../Action/action';

type TodoProps = {
    setName: (a: string) => void;
    reduxList: Array<String>;
};
type TodoState = {
    name: String;
};
class ToDoClass extends Component<TodoProps, TodoState> {
    state = {
        name: "",
    };
    handleClick = () => {
        if (!this.state.name.trim()) {
            alert("Empty text can't be added into redux");
        }
        this.props.setName(this.state.name);
        this.setState({ name: "" });
    };
    render() {
        return (
            <div style={{ display: 'flex' }}>
                <div style={{ margin: 'auto' }}>
                    <h3 style={{ marginTop: '60px' }}>Hi i am todo</h3>
                    <input
                        style={{ padding: '10px' }}
                        value={this.state.name}
                        type="text"
                        placeholder="enter your list value here! "
                        onChange={(e) => this.setState({ name: e.target.value })}
                    />
                    <br />
                    <button
                        type="button"
                        onClick={this.handleClick}
                        style={{ height: 45, width: 120, backgroundColor: "blue", color: 'white', marginTop: '10px', fontSize: '20px', borderRadius: '4px' }}
                    >
                        Submit
                    </button>
                    <br />
                    {this.props.reduxList.map((item, index) => (
                        <div key={index} style={{ fontSize: '20px' }}>
                            {item} <br />
                        </div>
                    ))}
                </div>
            </div>
        );
    }
}
function mapStateToProps(state: any) {
    return { reduxList: state.nameList };
}
export default connect(mapStateToProps, { setName })(ToDoClass);
