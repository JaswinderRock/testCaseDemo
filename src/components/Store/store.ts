import { createStore } from 'redux';
import { cakeReducer } from '../Reducer/cakeReducer';
import reducer from '../Reducer/incrementReducer';

export const incrementStore = createStore(reducer)
export const Store = createStore(cakeReducer);