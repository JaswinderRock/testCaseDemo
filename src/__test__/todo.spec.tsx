import React from 'react';
import { mount, shallow } from "enzyme"
import ToDoClass from '../components/Screen/ToDoClass';
import { Provider } from 'react-redux';
import { Store } from '../components/Store/store';

const setUp = () => {
    const component = mount(<Provider store={Store}>
        <ToDoClass />
    </Provider>);
    return component;
}

describe("Render TodoClass", () => {
    let component: any;
    const wrapper = setUp();
    beforeEach(() => {
        component = setUp();
    }),
    test("TodoList test is render  ", () => {
        const component = setUp()
        expect(component).toMatchSnapshot()
    })
    test('renders input and submit button', () => {
        expect(wrapper.find('input'));
        expect(wrapper.find('button[type="button"]'));
    })
    test('serach for label', () => {
        const wrapper = setUp();
        expect(wrapper.find("h3").text()).toContain("Hi i am todo");
    })

})