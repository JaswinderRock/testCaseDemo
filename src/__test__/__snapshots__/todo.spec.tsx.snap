// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render TodoClass TodoList test is render   1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ToDoClass)>
    <ToDoClass
      reduxList={Array []}
      setName={[Function]}
    >
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <div
          style={
            Object {
              "margin": "auto",
            }
          }
        >
          <h3
            style={
              Object {
                "marginTop": "60px",
              }
            }
          >
            Hi i am todo
          </h3>
          <input
            onChange={[Function]}
            placeholder="enter your list value here! "
            style={
              Object {
                "padding": "10px",
              }
            }
            type="text"
            value=""
          />
          <br />
          <button
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "blue",
                "borderRadius": "4px",
                "color": "white",
                "fontSize": "20px",
                "height": 45,
                "marginTop": "10px",
                "width": 120,
              }
            }
            type="button"
          >
            Submit
          </button>
          <br />
        </div>
      </div>
    </ToDoClass>
  </Connect(ToDoClass)>
</Provider>
`;
